aten/src/ATen/native/Normalization.cpp

```diff
@@ -416,6 +417,22 @@ std::tuple<Tensor, Tensor, Tensor, Tensor, int64_t> _batch_norm_impl_index(
   const Tensor& running_var = c10::value_or_else(running_var_opt, [] {return Tensor();});
 
   auto num_features = input.sizes()[1];
+
+  if (input.numel()==0){
```

Comment:
can you please leave a comment that under normal conditions this should never happen, and you will come here only from symbolic script.
Also, to delete this once you no longer rely on symbolic script and switch to lazy tensor.

Reply:
I had a long conversation on this topic with Alban offline.

We don't want exposed logic that would ended up with control flow in autodiff, because it breaks fusion :(, that defeats all the good things with supporting BN in autodiff.

This is not only for symbolic script. In eager mode you would also come here (we moved the short cut from `batch_norm` here, because we don't want to duplicate it). Arguably we could put that comment in `_batch_norm_impl_index_backward`, but in face you'll only go to that function through `symbolic_script`, so that seems to be a redundant comment.

Reply:
Looking at the stripped `batch_norm`, I think it makes sense to update the API so we no longer need `_batch_norm_impl_index`.
This makes `symbolic_script.cpp` more straightforward, so that we don't have to rewrite `batch_norm` with a different operation `_batch_norm_impl_index`, which feels pretty awkward.

I know updating the API seems to be a big hammer. Any comment on that?